From acbf7e15c9290e0891a6b6b5ce6e81bbaa77ce5a Mon Sep 17 00:00:00 2001
From: Johannes Weberhofer <jweberhofer@weberhofer.at>
Date: Sat, 11 Apr 2020 10:33:55 +0200
Subject: [PATCH] Drop of supplementary group IDs

Fix related to POS36-C and rpmlint error "missing-call-to-setgroups-before-setuid".
---
 src/apps/relay/mainrelay.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/apps/relay/mainrelay.c b/src/apps/relay/mainrelay.c
index 6710da6d..5a21cfaa 100644
--- a/src/apps/relay/mainrelay.c
+++ b/src/apps/relay/mainrelay.c
@@ -2061,6 +2061,7 @@ static void set_network_engine(void)
 
 static void drop_privileges(void)
 {
+	setgroups(0, NULL);
 	if(procgroupid_set) {
 		if(getgid() != procgroupid) {
 			if (setgid(procgroupid) != 0) {
